microfit/histogram/run_hist_generator.py

```diff
@@ -72,7 +73,9 @@ class is used.
             Additional keyword arguments that are passed to the MC histogram generator on initialization.
         """
         self.data_pot = data_pot
-        query = self.get_selection_query(selection, preselection)
+        extraq = None
+        if showdata == False: extraq = ["bnbdata==0"]
+        query = self.get_selection_query(selection, preselection,extraq)
```

Comment:
A better way to do this is to set the `HistogramGenerator` that makes the data histograms to `None`. This is already how the absence of data is handled (i.e. when we don't load the data in the first place).